src/KurrentDB.Client/Streams/KurrentDBClient.Metadata.cs

```diff
@@ -41,37 +60,111 @@ public async Task<StreamMetadataResult> GetStreamMetadataAsync(string streamName
-			TimeSpan? deadline = null, UserCredentials? userCredentials = null,
-			CancellationToken cancellationToken = default) {
-			var options = Settings.OperationOptions.Clone();
-			configureOperationOptions?.Invoke(options);
+		public Task<IWriteResult> SetStreamMetadataAsync(
+			string streamName,
+			StreamState expectedState,
+			StreamMetadata metadata,
+			SetStreamMetadata? operationOptions = null,
+			CancellationToken cancellationToken = default
+		) {
+			operationOptions ??= new SetStreamMetadata();
+			operationOptions.With(Settings.OperationOptions);
 
-			return SetStreamMetadataInternal(metadata, new AppendReq {
-				Options = new AppendReq.Types.Options {
-					StreamIdentifier = SystemStreams.MetastreamOf(streamName)
-				}
-			}.WithAnyStreamRevision(expectedState), options, deadline, userCredentials, cancellationToken);
+			return SetStreamMetadataInternal(
+				metadata,
+				new AppendReq {
+					Options = new AppendReq.Types.Options {
+						StreamIdentifier = SystemStreams.MetastreamOf(streamName)
+					}
+				}.WithAnyStreamRevision(expectedState),
+				operationOptions,
+				cancellationToken
+			);
 		}
 
-		private async Task<IWriteResult> SetStreamMetadataInternal(StreamMetadata metadata,
+		async Task<IWriteResult> SetStreamMetadataInternal(
+			StreamMetadata metadata,
 			AppendReq appendReq,
-			KurrentDBClientOperationOptions operationOptions,
-			TimeSpan? deadline,
-			UserCredentials? userCredentials,
-			CancellationToken cancellationToken) {
-
+			SetStreamMetadata operationOptions,
+			CancellationToken cancellationToken
+		) {
 			var channelInfo = await GetChannelInfo(cancellationToken).ConfigureAwait(false);
-			return await AppendToStreamInternal(channelInfo, appendReq, new[] {
-				new EventData(Uuid.NewUuid(), SystemEventTypes.StreamMetadata,
-					JsonSerializer.SerializeToUtf8Bytes(metadata, StreamMetadataJsonSerializerOptions)),
-			}, operationOptions, deadline, userCredentials, cancellationToken).ConfigureAwait(false);
+			return await AppendToStreamInternal(
+				channelInfo,
+				appendReq,
+				[
+					new MessageData(
+						SystemEventTypes.StreamMetadata,
+						JsonSerializer.SerializeToUtf8Bytes(metadata, StreamMetadataJsonSerializerOptions)
+					)
+				],
+				operationOptions,
+				cancellationToken
+			).ConfigureAwait(false);
+		}
+	}
+
+	public static class KurrentDBClientMetadataObsoleteExtensions {
+		/// <summary>
+		/// Asynchronously reads the metadata for a stream
+		/// </summary>
+		/// <param name="dbClient"></param>
+		/// <param name="streamName">The name of the stream to read the metadata for.</param>
+		/// <param name="deadline"></param>
+		/// <param name="userCredentials">The optional <see cref="UserCredentials"/> to perform operation with.</param>
+		/// <param name="cancellationToken">The optional <see cref="System.Threading.CancellationToken"/>.</param>
+		/// <returns></returns>
+		public static Task<StreamMetadataResult> GetStreamMetadataAsync(
```

Comment:
Was the intent for these to carry an [<Obsolete(...)>] attribute?

Reply:
Yes, good call! I forgot to mark them as obsolete. DId that in https://github.com/EventStore/EventStore-Client-Dotnet/pull/338/commits/c916dace6edce4cde6b480196d5741374defa475.